ophyd/v2/core.py

```diff
@@ -80,15 +80,19 @@ def done(self) -> bool:
         return self.task.done()
 
     @property
-    def success(self) -> bool:
-        assert self.done, "Status has not completed yet"
-        try:
-            self.task.result()
-        except (Exception, asyncio.CancelledError):
-            logging.exception("Failed status")
-            return False
+    def success(self) -> Optional[bool]:
+        # TODO find a better way to push exception upt to bluesky
+        # isinstance(obj, Satus) calls hasattr(obj, success) which causes the property getter to run
+        if self.done:
+            try:
+                self.task.result()
+            except (Exception, asyncio.CancelledError):
+                logging.exception("Failed status")
+                return False
+            else:
+                return True
         else:
-            return True
+            pass
```

Comment:
Should this be `False`?  If in has not finished it can not yet be successful.

Reply:
Yeah `False` makes sense will change it now.